### PR TITLE
feat(TableV2): new component

### PR DESCRIPTION
**Pull request by fabienhebert**
## Type

- Feature

### Summarise concisely:

Add a new component TableV2. The endgoal is to merge `Table (v1)`, `List (v1) variant table` & `List (v1) variant explorer` in this new component.

Table (V1) didn't supported ordering & selectable rows

### CODE REVIEW

File: packages/ui/src/components/ListV2/HeaderCell.tsx [hunk 1]
```diff
@@ -72,16 +72,27 @@ export const HeaderCell = ({
...
-        onOrder
-          ? () => onOrder(order === 'ascending' ? 'desc' : 'asc')
+      className={className}
+      onClick={handleOrder}
+      onKeyDown={
+        handleOrder
+          ? event => {
+              if (event.key === ' ') {
+                handleOrder()
+                event.preventDefault()
```
Comment (matthprost) [inline packages/ui/src/components/ListV2/HeaderCell.tsx:89]: I don't think we need the preventDefault 🤔

File: packages/ui/src/components/TableV2/HeaderCell.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  ${({ width, maxWidth, minWidth }) => `
+    ${width ? `width: ${width};` : ''}
+    ${maxWidth ? `max-width: ${maxWidth};` : ''}
+    ${minWidth ? `min-width: ${minWidth};` : ''}
+  `}
+
+  padding: ${({ theme }) => theme.space['1']};
+`
+
+const HeaderCellContent = styled('div')`
```
Comment (matthprost) [inline packages/ui/src/components/TableV2/HeaderCell.tsx:57]: Why not using a `Text` with prop `as="div"` so you don't need to add CSS related to text style

File: packages/ui/src/components/TableV2/HeaderCell.tsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    <StyledHeaderCell
+      aria-sort={order}
+      className={className}
+      onClick={handleOrder}
+      onKeyDown={
+        handleOrder
+          ? event => {
+              if (event.key === ' ') {
+                handleOrder()
+                event.preventDefault()
```
Comment (matthprost) [inline packages/ui/src/components/TableV2/HeaderCell.tsx:112]: Same here I don't think we need preventDefault it should work without

**Review by matthprost — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/ListV2/HeaderCell.tsx [hunk 4]
```diff
@@ -72,16 +72,27 @@ export const HeaderCell = ({
...
       aria-sort={order}
-      onClick={
-        onOrder
-          ? () => onOrder(order === 'ascending' ? 'desc' : 'asc')
+      className={className}
+      onClick={handleOrder}
+      onKeyDown={
+        handleOrder
+          ? event => {
+              if (event.key === ' ') {
```
Comment (ModuloM) [inline packages/ui/src/components/ListV2/HeaderCell.tsx:87]: Maybe we should also map `enter` key?
As we use this as a button (https://www.w3.org/WAI/ARIA/apg/patterns/button/)

File: packages/ui/src/components/TableV2/HeaderCell.tsx [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+  return (
+    <StyledHeaderCell
+      aria-sort={order}
+      className={className}
+      onClick={handleOrder}
+      onKeyDown={
+        handleOrder
+          ? event => {
+              if (event.key === ' ') {
```
Comment (ModuloM) [inline packages/ui/src/components/TableV2/HeaderCell.tsx:110]: Maybe we should also map enter key?

**Comment by ModuloM** [comment]
Juste quick note on checkbox component, it doesn't listen to keydown `Enter` to change it's state (but not related to TableV2) 🚀

**Comment by matthprost** [comment]
> Juste quick note on checkbox component, it doesn't listen to keydown `Enter` to change it's state (but not related to TableV2) 🚀

@ModuloM This might seems strange but for changing state with accessibility you should use space bar instead of enter. If you check on native checkbox that's how you change it and enter doesn't work 😛

File: packages/ui/src/components/ListV2/HeaderCell.tsx [hunk 6]
```diff
@@ -72,20 +72,34 @@ export const HeaderCell = ({
...
       aria-sort={order}
-      onClick={
-        onOrder
-          ? () => onOrder(order === 'ascending' ? 'desc' : 'asc')
+      className={className}
+      onClick={handleOrder}
+      onKeyDown={
+        handleOrder
+          ? event => {
+              if (event.key === ' ' || event.key === 'Enter') {
```
Comment (matthprost) [inline packages/ui/src/components/ListV2/HeaderCell.tsx:87]:
```suggestion
              if (event.key === ' ') {
```

File: packages/ui/src/components/TableV2/HeaderCell.tsx [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+    : undefined
+
+  return (
+    <StyledHeaderCell
+      className={className}
+      onClick={handleOrder}
+      onKeyDown={
+        handleOrder
+          ? event => {
+              if (event.key === ' ' || event.key === 'Enter') {
```
Comment (matthprost) [inline packages/ui/src/components/TableV2/HeaderCell.tsx:103]:
```suggestion
              if (event.key === ' ') {
```
Screenshot 2023-05-03 at 10 32 43

Here is a nice tool to get the value of each keys: https://www.toptal.com/developers/keycode

File: packages/ui/src/components/TableV2/__stories__/Style.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+  { stripped: false, bordered: false },
+  { stripped: true, bordered: false },
+  { stripped: false, bordered: true },
+  { stripped: true, bordered: true },
+]
+
+export const Style: Story = () => (
+  <Stack gap={2}>
+    {combos.map(combo => (
+      <Stack gap={1}>
```
Comment (matthprost) [inline packages/ui/src/components/TableV2/__stories__/Style.stories.tsx:16]: A key is missing

**Review by matthprost — APPROVED** [review]

**Pull request merged**